src/java.base/share/classes/java/util/concurrent/CompletableFuture.java

```diff
@@ -2784,9 +2758,8 @@ public CompletionStage<T> minimalCompletionStage() {
      */
     public CompletableFuture<T> completeAsync(Supplier<? extends T> supplier,
                                               Executor executor) {
-        if (supplier == null || executor == null)
-            throw new NullPointerException();
-        executor.execute(new AsyncSupply<T>(this, supplier));
+        executor.execute(new AsyncSupply<T>(
```

Comment:
Probably missing `Objects.requireNonNull` to preserve `executor == null` check:

```suggestion
        Objects.requireNonNull(executor).execute(new AsyncSupply<T>(
```

Reply:
Thanks! Somehow we were missing a tck test case for this. I'll add one.

Reply:
Rechecking this, it doesn't need changing since there is already an implicit null check on executor.execute.